src/movie/List.js

```diff
@@ -0,0 +1,142 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { MovieCard } from './Card';
+import { genres, movieSortCritera, movieList } from './Const';
+
+export class MovieList extends Component {
+  constructor(props) {
+    super(props);
+    this.allMovies = movieList;
+
+    this.sortCriteriaList = movieSortCritera;
+    this.sortKey = this.sortCriteriaList[0].code;
+    const sortedMovies = this.sortMovie(this.sortKey, this.allMovies);
+    this.state = { activeGenre: genres[0], filteredMovies: sortedMovies };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.query !== this.props.query) {
+      this.filterMoviesByQuery();
+    }
+  }
+
+  /** Filter movie by given query text, then reset the genre to `All` and finally sort it */
+  filterMoviesByQuery() {
+    const query = this.props.query.toLowerCase();
+    const filteredMovies = !query
+      ? this.allMovies
+      : this.allMovies.filter((movie) => movie.title.toLowerCase().includes(query));
+    const sortedMovies = this.sortMovie(this.sortKey, filteredMovies);
+    this.setState({ activeGenre: 'All', filteredMovies: sortedMovies });
```

Comment:
you can name this variable as **filteredMovies** and then setState will be more shorter:
```
this.setState({
  activeGenre: 'All',
  filteredMovies,
});
```